python/src/pymain.cc

```diff
@@ -23,97 +13,9 @@ static void SetUpLocalEnvironment() {
-                boost::python::stl_input_iterator<PythonLightField>(listOfLightFields),
-                boost::python::stl_input_iterator<PythonLightField>());
-
-        std::vector<lightdb::LightFieldReference> lightFields;
-        std::transform(pythonLightFields.begin(), pythonLightFields.end(), std::back_inserter(lightFields),
-                [](PythonLightField lightField) -> lightdb::LightFieldReference { return lightField.query(); });
-
-        return PythonLightField(_lightField.Union(lightFields));
-    }
-
-    PythonLightField Discretize(const lightdb::Dimension dimension, double interval) {
-        return PythonLightField(_lightField.Discretize(dimension, lightdb::number(interval)));
-    }
-
-    PythonLightField Interpolate(lightdb::Dimension dimension) {
-        return PythonLightField(_lightField.Interpolate(dimension, lightdb::interpolation::Linear()));
-    }
-
-    PythonLightField Map(PyObject *udf, std::filesystem::path path) {
-        auto yolo = lightdb::extensibility::Load("yolo", path);
-        return PythonLightField(_lightField.Map(yolo));
-    }
-
-    PythonLightField Map(lightdb::functor::unaryfunctor functor) {
-        return PythonLightField(_lightField.Map(functor));
-    }
-
-    PythonLightField Encode() {
-        return PythonLightField(_lightField.Encode());
-    }
-
-    PythonLightField Save(const std::string &fileName) {
-        return PythonLightField(_lightField.Save(static_cast<std::filesystem::path>(fileName)));
-    }
-
-    PythonLightField Store(const lightdb::catalog::Catalog &catalog, const std::string &name) {
-        return PythonLightField(_lightField.Store(name, catalog));
-    }
-
-    lightdb::LightFieldReference query() {
-        return _lightField;
-    }
-
-private:
-    lightdb::LightFieldReference _lightField;
-};
-
-PythonLightField (PythonLightField::*SelectPhi)(const lightdb::PhiRange&) = &PythonLightField::Select;
-PythonLightField (PythonLightField::*SelectTheta)(const lightdb::ThetaRange&) = &PythonLightField::Select;
-PythonLightField (PythonLightField::*SelectSpatiotemporal)(lightdb::SpatiotemporalDimension, const lightdb::SpatiotemporalRange&) = &PythonLightField::Select;
-PythonLightField (PythonLightField::*UnionOne)(PythonLightField&) = &PythonLightField::Union;
-PythonLightField (PythonLightField::*UnionMany)(boost::python::list&) = &PythonLightField::Union;
-PythonLightField (PythonLightField::*PythonMap)(PyObject*, std::filesystem::path) = &PythonLightField::Map;
-PythonLightField (PythonLightField::*FunctorMap)(lightdb::functor::unaryfunctor) = &PythonLightField::Map;
-
-
 static PythonLightField Load(const std::string &filepath, boost::python::dict string_options) {
-    PythonOptions opt(string_options);
-    return PythonLightField(lightdb::logical::Load(filepath, opt.options()));
+PythonOptions opt(string_options);
+return PythonLightField(lightdb::logical::Load(filepath, opt.options()));
```

Comment:
PythonOptions should have an implicit conversion to lightdb::options, not a member function.